src/controllers/authController.ts

```diff
@@ -93,5 +95,47 @@ const login = async (req: Request, res: Response): Promise<void> => {
     sendInternalErrorResponse(res, err);
   }
 };
+const updatePassword = async (req: Request, res: Response): Promise<void> => {
+  try {
+    const { oldPassword, newPassword } = req.body;
+
+    // Access decoded user information from the request object
+    const user = req.user as {
+      id: string;
+      password: string;
+    };
+
+    // Check if old password matches with the given one
+    const match = await passwordCompare(oldPassword, user.password);
+    if (!match) {
+      res.status(400).json({
+        ok: false,
+        message: 'The old password is incorrect!',
+      });
+      return;
+    }
+
+    // Generate salt and hash new password
+    const hashedNewPassword = await passwordEncrypt(newPassword);
```

Comment:
Dont we have to validate newPassword?

Reply:
You are right. @A-gent64  validate passwords